common/workunit/workunit.cpp

```diff
@@ -2988,10 +2971,20 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
                 case WUStateDebugRunning:
                 case WUStateBlocked:
                 case WUStateAborting:
-                    SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
-                    if (checkAbnormalTermination(wuid, ret, agent))
+                    if (queryDaliServerVersion().compare("2.1")>=0)
                     {
-                        return ret;
+                        SessionId agent = conn->queryRoot()->getPropInt64("@agentSession", -1);
+                        if((agent>0) && querySessionManager().sessionStopped(agent, 0))
+                        {
+                            conn->reload();
+                            // MCK - does checking state after sessionStopped reveal meaningful info ?
+                            WUState finalState = (WUState) getEnum(conn->queryRoot(), "@state", states);
+                            if ( (WUStateCompleted != finalState) &&
+                                 (WUStateFailed    != finalState) &&
+                                 (WUStateAborted   != finalState) )
```

Comment:
Instead of this code, would it simpler/cleaner to note that session stopped and loop (continue) instead?
Then the existing code would pick up valid (completed) states and return as normal.
Or if truly died, it would re-enter here, in which case you could say:
```
if (agentSessionStopped) // marked true on previous pass
    reportAbnormalTermination(wuid, ret, state, agent);
```